aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -228,17 +425,70 @@ module aptos_framework::coin {
         account_addr: address,
         amount: u64,
         dst_coin: &mut AggregatableCoin<CoinType>,
-    ) acquires CoinStore {
+    ) acquires CoinStore, CoinConversionMap, CoinInfo, PairedCoinType {
         // Skip collecting if amount is zero.
         if (amount == 0) {
             return
         };
 
-        let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
-        let coin = extract(&mut coin_store.coin, amount);
+        let coin_balance = coin_balance<CoinType>(account_addr);
+        let metadata = paired_metadata<CoinType>();
+        let (coin_amount_to_collect, fa_amount_to_collect) = if (coin_balance >= amount || option::is_none(
+            &metadata
+        )) (amount, 0) else (coin_balance, amount - coin_balance);
+        let coin = if (coin_amount_to_collect > 0) {
+            let coin_store = borrow_global_mut<CoinStore<CoinType>>(account_addr);
+            extract(&mut coin_store.coin, coin_amount_to_collect)
+        } else {
+            zero()
+        };
+        if (fa_amount_to_collect > 0) {
+            let store_addr = primary_fungible_store::primary_store_address(account_addr, option::extract(&mut metadata));
+            let fa = fungible_asset::withdraw_internal(store_addr, fa_amount_to_collect);
+            merge(&mut coin, fungible_asset_to_coin<CoinType>(fa));
+        };
         merge_aggregatable_coin(dst_coin, coin);
     }
 
+    fun maybe_convert_to_fungible_store<CoinType>(account: address) acquires CoinStore, CoinConversionMap, CoinInfo {
+        if (exists<CoinStore<CoinType>>(account)) {
+            if (features::coin_to_fungible_asset_migration_feature_enabled()) {
```

Comment:
could we eliminate the excessive nesting and just have a check to see if the feature is enabled?

Reply:
how?

Reply:
check if the feature is enabled first in an if, then you can just make another if or else if on the exists